src/app/shared/components/template/services/template-variables.service.ts

```diff
@@ -183,6 +183,11 @@ export class TemplateVariablesService extends AsyncServiceBase {
         return evaluator.fullExpression.replace(/`/gi, "");
       }
 
+      // Do not evaluate if the appropriate context is not available
```

Comment:
This took me ages to figure out! There may be a nicer solution, but this is why something like this is needed:

Even though we have the check lower down (line 393), without this additional check then the string gets evaluated as JS, without the dynamic evaluator value being replaced. This meant that my string `@item._index + 1` was being evaluated as JS before `itemContext` was available, evaluating to `@item._index1` because it was treated as the addition of a string and a number, `"@item._index" + 1`. Then, when the expression was re-evaluated after the data-items component had added the `itemContext`, `@item._index1` was evaluated with the value of `@item_index` replaced. For the item at index `0`, this actually meant the expression evaluated as `01`, i.e. `1`, and so for that one case the evaluation actually gave the expected value, which appeared as a false positive in my testing...

Reply:
Really useful to catch, it would be good to include such things in spec test if possible